source/_integrations/cast.markdown

```diff
@@ -48,17 +48,122 @@ Home Assistant Cast requires your Home Assistant installation to be accessible v
+
+The media needs to be accessible via HTTP(S). Chromecast devices do not support other protocols like DLNA or playback from an SMB file share.
+
+You can play MP3 streams like net radios, FLAC files or videos from your local network with the `media_player.play_media` service, as long as the media is accessible via HTTP(S). You need to set the `media_content_id` to the media URL and `media_content_type` to a matching content type.
+
+```yaml
+# Play a video file from the local network:
+service: media_player.play_media
+target:
+  entity_id: media_player.chromecast
+data:
+  media_content_type: "video"
+  media_content_id: "http://192.168.0.100/movies/sample-video.mkv"
+```
+
+```yaml
+# Show a jpeg image:
+service: media_player.play_media
+target:
+  entity_id: media_player.chromecast
+data:
+  media_content_type: "image/jpeg"
+  media_content_id: "http://via.placeholder.com/1024x600.jpg/0B6B94/FFFFFF/?text=Hello,%20Home%20Assistant!"
+```
+
+Extra media metadata (for example title, subtitle, artist or album name) can be passed into the service and that will be shown on the Chromecast display.
+For the possible metadata types and values check [Google cast documentation > MediaInformation > metadata field](https://developers.google.com/cast/docs/reference/messages#MediaInformation).
+
+```yaml
+# Play a movie from the internet, with extra metadata provided:
+service: media_player.play_media
+target:
+  entity_id: media_player.chromecast
+data:
+  media_content_type: "video/mp4"
+  media_content_id: "http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4"
+  extra: 
+    metadata: 
+      metadataType: 1
+      title: "Big Buck Bunny"
+      subtitle: "By Blender Foundation, Licensed under the Creative Commons Attribution license"
+      images:
+        - url: "https://peach.blender.org/wp-content/uploads/watchtrailer.gif"
+```
+
+```yaml
+# Play a netradio, with extra metadata provided:
+service: media_player.play_media
+target:
+  entity_id: media_player.chromecast
+data:
+  media_content_type: "audio/mp3"
+  media_content_id: "http://stream.tilos.hu:8000/tilos" 
+  extra: 
+    metadata: 
+      metadataType: 3
+      title: "Radio TILOS"
+      artist: "LIVE"
+      images:
+        - url: "https://tilos.hu/images/kockalogo.png"
+```
+
+### Casting with other apps
+
+It's possible to play with other apps than the default media receiver.
+To do so, `media_content_type` should be set to `cast`, and `media_content_id` should be a JSON dict with parameters for the app, including the app name.
+
+### BubbleUPNP
+
+The BubbleUPNP app has similar functionality to the built in Default Media Receiver app, and can be used as a backup if the default app fails to play the media.
```

Comment:
Is this an app that is always built in with Chromecast devices and/or Nest Hubs? And if so, should we mention this?

Reply:
There is no user control to manage or install apps on a Chromecast device. Chromecast apps are installed/fetched automatically when it is started/requested.

Reply:
It was a good comment though, and the concept of "Chromecast Apps" is quite confusing because it's never seen by the user, he or she just "casts" something to the device.
Should we add a really brief introduction trying to explain what's going on?